File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end());
        vector<vector<int>> mergedNonOverlappingIntervals;
        int len= 0, n=(int)intervals.size();
        for(int i=0;i<n;++i){
            if(len==0 || 
                mergedNonOverlappingIntervals[len-1][1]<intervals[i][0]){
                mergedNonOverlappingIntervals.push_back(intervals[i]);
                ++len;
            }
            else{
                mergedNonOverlappingIntervals[len-1][1]= 
                    max(mergedNonOverlappingIntervals[len-1][1], intervals[i][1]);
            }
        }
        return mergedNonOverlappingIntervals;
    }
};

File: 56-merge-intervals/merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        int n= intervals.length;
        Arrays.sort(intervals, (a,b)->a[0]-b[0]); 
        //Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        int mergedIdx= 0;
        for(int i=1;i<n;i++){
            if(intervals[i][0]<=intervals[mergedIdx][1]){
                intervals[mergedIdx][1]= Math.max(intervals[mergedIdx][1], intervals[i][1]);
            }
            else{
                intervals[++mergedIdx] = intervals[i];
            }
        }
        int[][] mergedIntervals = new int[mergedIdx+1][2];
        for(int i=0;i<=mergedIdx;i++) mergedIntervals[i] = intervals[i];
        return mergedIntervals;
    }
}